core/camera/src/main/java/com/google/jetpackcamera/core/camera/CameraSession.kt

```diff
@@ -150,80 +150,132 @@ internal suspend fun runSingleCameraSession(
+                if (videoCaptureUseCase != null) {
+                    val videoQuality = getVideoQualityFromResolution(
+                        videoCaptureUseCase.resolutionInfo?.resolution
                     )
-                }
-                launch {
-                    currentCameraState.update { old ->
-                        old.copy(
-                            videoQualityInfo = VideoQualityInfo(
-                                videoQuality,
-                                getWidthFromCropRect(videoCaptureUseCase.resolutionInfo?.cropRect),
-                                getHeightFromCropRect(videoCaptureUseCase.resolutionInfo?.cropRect)
-                            )
+                    if (videoQuality != sessionSettings.videoQuality) {
+                        Log.e(
+                            TAG,
+                            "Failed to select video quality: $sessionSettings.videoQuality. " +
+                                "Fallback: $videoQuality"
                         )
                     }
+                    launch {
+                        currentCameraState.update { old ->
+                            old.copy(
+                                videoQualityInfo = VideoQualityInfo(
+                                    videoQuality,
+                                    getWidthFromCropRect(
+                                        videoCaptureUseCase.resolutionInfo?.cropRect
+                                    ),
+                                    getHeightFromCropRect(
+                                        videoCaptureUseCase.resolutionInfo?.cropRect
+                                    )
+                                )
+                            )
+                        }
+                    }
                 }
-            }
 
-            applyDeviceRotation(currentTransientSettings.deviceRotation, useCaseGroup)
-            setZoomScale(camera, 1f)
-            processTransientSettingEvents(
-                camera,
-                useCaseGroup,
-                currentTransientSettings,
-                transientSettings
-            )
+                // update camerastate to mirror current zoomstate
+
+                launch {
+                    camera.cameraInfo.zoomState
+                        .asFlow()
+                        .filterNotNull()
+                        .distinctUntilChanged()
+                        .onCompletion {
+                            // reset current camera state when changing cameras.
+                            currentCameraState.update { old ->
+                                old.copy(
+                                    zoomRatios = emptyMap(),
+                                    linearZoomScales = emptyMap()
+                                )
+                            }
+                        }
+                        .collectLatest { zoomState ->
+                            // TODO(): remove checks after buggy zoomState is fixed
```

Comment:
Add a link to the android issue tracker with the bug for this condition inside the parenthesist of `TODO()`.